include/swift/Remote/MetadataReader.h

```diff
@@ -195,25 +193,43 @@ class MetadataReader {
   /// amounts of data when we encounter corrupt values for sizes/counts.
   static const uint64_t MaxMetadataSize = 1048576; // 1MB
 
-  /// Define a has function for a std::pair<StoredPointer, bool>
-  struct TypeCacheKeyHash {
-    std::size_t operator()(const std::pair<StoredPointer, bool> &x) const {
-      return llvm::hash_combine(x.first, x.second);
+  /// The dense map info for a std::pair<StoredPointer, bool>.
+  struct DenseMapInfoTypeCacheKey {
+    using Pair = std::pair<StoredPointer, bool>;
+    using StoredPointerInfo = llvm::DenseMapInfo<StoredPointer>;
```

Comment:
This shouldn't be necessary.  DenseMapInfo has already a generic specialization for std::pair as long as both elements are supported. You should only need to define this for StoredPointer if at all.

Reply:
> DenseMapInfo has already a generic specialization for std::pair as long as both elements are supported

`bool` is not supported since it can't implement `getEmptyKey` or `getTombstoneKey`. This implementation uses only the `StoredPointer`'s implementation for both those functions.